DataConnectors/GCP/Terraform/sentinel_resources_creation/GCPCloudIDSLogSetup/GCPCloudIDSLogSetup.tf

```diff
@@ -0,0 +1,106 @@
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "3.73.0"
+    }
+  }
+
+  required_version = ">= 0.15.0"
+}
+
+variable "project-id" {
+  type        = string
+  description = "Enter your project ID"
+}
+
+variable "topic-name" {
+  type        = string
+  default     = "sentinelcloudids-topic"
+  description = "Name of existing topic"
+}
+
+variable "organization-id" {
+  type        = string
+  default     = ""
+  description = "Organization id"
+}
+
+data "google_project" "project" {
+  project_id = var.project-id
+}
+
+resource "google_project_service" "enable-logging-api" {
+  service = "logging.googleapis.com"
+  project = data.google_project.project.project_id
+}
+
+resource "google_pubsub_topic" "sentinelcloudids-topic" {
+  count   = "${var.topic-name != "sentinelcloudids-topic" ? 0 : 1}"
+  name    = var.topic-name
+  project = data.google_project.project.project_id
+}
+
+resource "google_pubsub_subscription" "sentinel-subscription" {
+  project = data.google_project.project.project_id
+  name    = "sentinel-subscription-CLOUDIDSlogs"
+  topic   = var.topic-name
+  depends_on = [google_pubsub_topic.sentinelcloudids-topic]
+}
+
+resource "google_logging_project_sink" "sentinel-sink" {
+  project    = data.google_project.project.project_id
+  count      = var.organization-id == "" ? 1 : 0
+  name       = "CLOUDIDS-logs-sentinel-sink"
+  destination = "pubsub.googleapis.com/projects/${data.google_project.project.project_id}/topics/${var.topic-name}"
+  depends_on = [google_pubsub_topic.sentinelcloudids-topic]
+
+  filter = "protoPayload.serviceName="ids.googleapis.com" OR (resource.type="ids.googleapis.com/Endpoint") OR (resource.type="ids.googleapis.com/Endpoint" AND jsonPayload.alert_severity=("INFORMATIONAL" OR "LOW" OR "MEDIUM" OR "HIGH" OR "CRITICAL"))"
```

Comment:
getting error while running the script 

![image](https://github.com/user-attachments/assets/632e3bc7-b5d9-4070-b05a-37f44bfbd3eb)